### PR TITLE
Issue696 heat pump improvements and minor bugs

### PR DESCRIPTION
**Pull request by alexanderAKU**
> Thank you for making a Pull Request to AixLib!

> Please consider these points before submitting:
> - Mention the issue that this Pull Request intends to solve, and briefly describe what the main changes are.
> - Make sure all new code follows our [Modelica Guidelines](https://github.com/RWTH-EBC/AixLib/wiki/Modelica-guidelines)
> - Make sure all new models include documentation as described in the [Documentation Wiki-Page](https://github.com/RWTH-EBC/AixLib/wiki/Documentation) and are demonstrated by suitable examples

> If you know someone who could review your code, please use the *Reviewer* function on the right to ask for a review and also assign the PR to them. If not, no worries, we will find someone for you.

> Thanks again for your PR! You can delete these first lines before submitting your PR.

### CODE REVIEW

**Comment by alexanderAKU** [comment]
@DaJansenGit I can restructure the example. We could also delete the real and boolean input completely.

**Comment by alexanderAKU** [comment]
@DaJansenGit there is another bug: The start temperature for the condensor can't be set if the heat transfer is deactivated. However, the start temperature is independent from the heat transfer.

**Comment by DaJansenGit** [comment]
> @DaJansenGit I can restructure the example. We could also delete the real and boolean input completely.

Yes, from my site we can remove these inputs. The sigbus should be used anyway.

**Comment by DaJansenGit** [comment]
> @DaJansenGit there is another bug: The start temperature for the condensor can't be set if the heat transfer is deactivated. However, the start temperature is independent from the heat transfer.

Thanks for fixing this bug.
If you remove the real inputs and remake the examples we are ready to merge ;) Thanks in advance!

**Comment by alexanderAKU** [comment]
@DaJansenGit I removed the real/boolean connectors in the HeatPumpSystem model that connect the heatpump with the controller

**Comment by PMehrfeld** [comment]
Closes #696

**Comment by PMehrfeld** [comment]
Wait until ToDo is checked in [this comment](https://github.com/RWTH-EBC/AixLib/issues/696#issuecomment-559016294).

**Comment by FWuellhorst** [comment]
@DaJansenGit : I implemented all required changes into the branch and also merged the latest development into this branch. Let's see if the regression tests will fail based on the new initialization method.

**Comment by DaJansenGit** [comment]
Thanks @FWuellhorst. Due to the changes in the initialization the regression tests are failing. I will update the reference results.

**Comment by DaJansenGit** [comment]
@FWuellhorst I updated the reference results. Let's see if I had the correct settings and the tests are passing. Buildingspy might used Dymola2020x on my computer, so let's hope this doesn't lead to diffs in the reference results ;)

**Comment by FWuellhorst** [comment]
Thanks for updating the reference results! I assume the changed initial conditions lead to the error in the unit-tests. 
From the generated log:
```
Warning: Assuming fixed start value for the continuous states:
    heatPumpSystem.heatPump.con.heatCap.T(start = heatPumpSystem.heatPump.con.TCap_start)
    heatPumpSystem.heatPump.con.vol.dynBal.U(start = 666485.09184)
    heatPumpSystem.heatPump.eva.heatCap.T(start = heatPumpSystem.heatPump.eva.TCap_start)
    heatPumpSystem.heatPump.eva.vol.dynBal.U(start = 166621.27296)
```
Other passing tests for Systems have all initial conditions specified. 

I know that we can fix this by setting energyDynamics==FixedInitial. 
However i am not sure whether this is the correct way to deal with the problem.
Furthermore, it's strange that the changes only affect heat pump system, while the HeatPump test passes.

**Comment by PMehrfeld** [comment]
@FWuellhorst: I did not come across this issue in your mentioned issue. But I think I solved this issue in the HeatPumpSystem example once.  
If I remember correctly, energy dynamics must be set individually in pumps, fluid volume of HX and dry mass capacities of HX.

**Comment by PMehrfeld** [comment]
I created a new branch from `development` to update this branch. See #878 .

Mainly, I came across the issue that the `DataBase` package contains now a `ThermalMachines` sub-package including `HeatPump` and `Chiller` that are also on the level of `DataBase`.

This seems to happened with commit d369f1a20a82d6a10ada6eb4a7411cdebbb17ac7.  
A bit about the history:
1. In a pretty old version of AixLib (v0.4.0) we had **only** `DataBase.HeatPump`.
1. In current `master` and `development` we have **only** `DataBase.ThermalMachines`.
1. I the first commit of this branch and, thus, in the commit before the above mentioned one we had `DataBase.HeatPump` and `DataBase.Chiller` but **no** `DataBase.ThermalMachines` ([Link](https://github.com/RWTH-EBC/AixLib/tree/b6f01afd3543d6bcb04e3a74e004ace07c647a76/AixLib/DataBase))
1. In the commit mentioned above we had all three folders (**including** `DataBase.ThermalMachines`.) ([Link](https://github.com/RWTH-EBC/AixLib/tree/d369f1a20a82d6a10ada6eb4a7411cdebbb17ac7/AixLib/DataBase))

We have no conversion script linking to `...DataBase.ThermalMachines...`. This means, there was a non-backwards-compatible change.

**Comment by PMehrfeld** [comment]
- [ ] One issue might be this one. Is `n` or `N` correct? This belongs to the file(s) `HeatPump.PerformanceData.LookUpTable XX .mo`  

image

**Comment by PMehrfeld** [comment]
I added all 3 packages again in d31b7e3. Hopefully without any loss of data or changes made at any point.

**Comment by PMehrfeld** [comment]
@FWuellhorst: You probably know the records and functions the best. Would you please check in the latest commit of this branch the differences between 
- `DataBase.HeatPump` and `DataBase.ThermalMachines.HeatPump`
- `DataBase.Chiller` and `DataBase.ThermalMachines.Chiller`

Change everything to the and keep only one sub-package each.  
My suggestion is to 
- [ ] keep `DataBase.HeatPump` and `DataBase.Chiller` and 
- [ ] add a conversion script for `DataBase.ThermalMachines.X` --> `DataBase.X`.

**Comment by FWuellhorst** [comment]
While explaing the model to another student i came across further improvements. The model `InverterControlledHP` works, only the PI-Values are not propagated. Furthermore the constrainedby value in `PartialHPSystem` was wrong. I fixed both.

**Comment by FWuellhorst** [comment]
> 
> 
>     * [ ]  One issue might be this one. Is `n` or `N` correct? This belongs to the file(s) `HeatPump.PerformanceData.LookUpTable XX .mo`
> 
> 
> image

I could not find any issue with that in the latest commit, everything is correctly refactored to `sigBus.n`

**Comment by DaJansenGit** [comment]
When opening the AixLib i get the following error:
```<innerCycle modification> (line 196, column 69: D:/01_GitHub/AixLibDevelopment/AixLib/Fluid/BaseClasses/PartialReversibleThermalMachine.mo)

Ambiguous modifications replaceable AixLib.Fluid.BaseClasses.PartialInnerCycle innerCycle constrainedby AixLib.Fluid.BaseClasses.PartialInnerCycle "Blackbox model of refrigerant cycle of a thermal machine" annotation (Placement(transformation(extent={{-27,-26},{27,26}}, rotation=90, origin={0,-1}))) and innerCycle.
Multiple modification of the same component is prohibited.
It is prohibited to modify subcomponents of a final, replaceable or redeclare modified component C except inside the modification of C.```

@FWuellhorst could you have a look at this?

**Comment by FWuellhorst** [comment]
While solving the SF error for the NDTable models, i noticed that the replaceable `PerData` Models are misleading.
grafik

We either have to change the description OR  (which i am in favour of) make `AixLib.DataBase.Chiller.PerformanceData.BaseClasses.PartialPerformanceData` standalone and not extend from `AixLib.DataBase.HeatPump.PerformanceData.BaseClasses.PartialPerformanceData`.

If this is not wanted by you, just remove the last commit (434ce85).

File: AixLib/DataBase/Chiller/PerformanceData/BaseClasses.mo [hunk 1]
```diff
@@ -2,8 +2,31 @@ within AixLib.DataBase.Chiller.PerformanceData;
...
+          rotation=270,
+          origin={-80,-110})));
+    AixLib.Controls.Interfaces.ThermalMachineControlBus sigBus
+      "Bus-connector used in a thermal machine" annotation (Placement(
+          transformation(
+          extent={{-15,-14},{15,14}},
+          rotation=0,
+          origin={1,104})));
+    Modelica.Blocks.Interfaces.RealOutput QEva(final unit="W", final displayUnit="kW")
+                                                                           "Heat flow rate through Condenser"  annotation (Placement(
```
Comment (PMehrfeld) [inline AixLib/DataBase/Chiller/PerformanceData/BaseClasses.mo:23]: Isn't this then "Heat flow rate through evaporator"?

File: AixLib/DataBase/Chiller/PerformanceData/BaseClasses.mo [hunk 2]
```diff
@@ -2,8 +2,31 @@ within AixLib.DataBase.Chiller.PerformanceData;
...
+          transformation(
+          extent={{-10,-10},{10,10}},
+          rotation=270,
+          origin={80,-110})));
+  protected
+    parameter Real scalingFactor=1 "Scaling factor of heat pump";
```
Comment (PMehrfeld) [inline AixLib/DataBase/Chiller/PerformanceData/BaseClasses.mo:29]: This model now belongs to a chiller and not heat pump ;)

**Comment by PMehrfeld** [comment]
CIs have been successful. Since review comments of @DaJansenGit were addressed, I now merge this branch.

Thanks to all for this great work!!

**Pull request merged**